### PR TITLE
Rados df converted values into mixed units

### PR DESCRIPTION
**Pull request by souravchk**
* https://jira.fkinternal.com/browse/DFORTYTWO-1671

### CODE REVIEW

File: src/tools/rados/rados.cc [hunk 1]
```diff
@@ -52,6 +52,20 @@ int rados_tool_sync(const std::map < std::string, std::string > &opts,
...
+ * are being printed with mixed units,
+ * the helper constant values used are
+ * defined below.
+ */
+#define BUFFER_SIZE 500
+#define BILLION 1000000000
+#define MILLION 1000000
+#define KILO 1000
+#define BY 1 
+#define KB BY * 1024
```
Comment (rollandsantimano) [inline src/tools/rados/rados.cc]: Braces required

File: src/tools/rados/rados.cc [hunk 2]
```diff
@@ -52,6 +52,20 @@ int rados_tool_sync(const std::map < std::string, std::string > &opts,
 #define STR(x) _STR(x)
 #define _STR(x) #x
 
+/* rados df -h command's metric values 
+ * are being printed with mixed units,
+ * the helper constant values used are
+ * defined below.
+ */
+#define BUFFER_SIZE 500
+#define BILLION 1000000000
```
Comment (rollandsantimano) [inline src/tools/rados/rados.cc]: Convert to use (MILLION * 1000), etc

File: src/tools/rados/rados.cc [hunk 3]
```diff
@@ -1487,10 +1556,14 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
...
     if (!formatter) {
-      printf("  total used    %12lld %12lld\n", (long long unsigned)tstats.kb_used,
-	     (long long unsigned)tstats.num_objects);
-      printf("  total avail   %12lld\n", (long long unsigned)tstats.kb_avail);
-      printf("  total space   %12lld\n", (long long unsigned)tstats.kb);
+      char buf_kb_used_total[BUFFER_SIZE],
+	   buf_num_objects_total[BUFFER_SIZE],
+	   buf_kb_avail_total[BUFFER_SIZE],
+	   buf_kb_total[BUFFER_SIZE];
+      printf("  total used    %12s %12s\n", convert_size(buf_kb_used_total,tstats.kb_used),
```
Comment (rollandsantimano) [inline src/tools/rados/rados.cc:1563]: check alignment with long pool names, such as in D42 prod clusters

File: src/tools/rados/rados.cc [hunk 4]
```diff
@@ -1128,6 +1142,52 @@ static int do_cache_flush_evict_all(IoCtx& io_ctx, bool blocking)
...
+      written = 1; break;
+    }
+  }
+  if (written == 0) {
+    snprintf(buf, BUFFER_SIZE, "%lld", count);
+  }
+  return buf;
+}
+
+char * convert_size(char *buf, long long size)
```
Comment (rollandsantimano) [inline src/tools/rados/rados.cc]: is 'size' signed or unsigned ?

File: src/tools/rados/rados.cc [hunk 5]
```diff
@@ -1128,6 +1142,52 @@ static int do_cache_flush_evict_all(IoCtx& io_ctx, bool blocking)
...
+{
+  const char* suffixes[] = {"Gb", "Mb", "Kb", "b"};
```
Comment (rollandsantimano) [inline src/tools/rados/rados.cc]: These should be GB not Gb -- Gigabytes vs Gigabits

File: src/tools/rados/rados.cc (hunk 2 above)
Comment (souravchk) [inline src/tools/rados/rados.cc]: done.

File: src/tools/rados/rados.cc (hunk 1 above)
Comment (souravchk) [inline src/tools/rados/rados.cc]: done

File: src/tools/rados/rados.cc (hunk 4 above)
Comment (souravchk) [inline src/tools/rados/rados.cc]: changed to unsigned, as the two metrics: size and count should not be negative!

File: src/tools/rados/rados.cc (hunk 5 above)
Comment (souravchk) [inline src/tools/rados/rados.cc]: done

File: src/tools/rados/rados.cc (hunk 3 above)
Comment (souravchk) [inline src/tools/rados/rados.cc]: Will check in the clusters here. Alignment can be taken up later?

File: src/tools/rados/rados.cc [hunk 6]
```diff
@@ -52,6 +52,20 @@ int rados_tool_sync(const std::map < std::string, std::string > &opts,
...
+/* rados df command's output metric values 
+ * are being printed with mixed units,
+ * the helper constant values used are
+ * defined below.
+ */
+#define BUFFER_SIZE 500
+#define BILLION (1000 * MILLION)
+#define MILLION (1000 * THOUSAND)
+#define THOUSAND 1000
+#define BY 1
```
Comment (rollandsantimano) [inline src/tools/rados/rados.cc]: is this short for BYTES ? in that case, make it explicit ...

Comment (souravchk) [inline src/tools/rados/rados.cc]: Yes. Ok sure.

**Pull request merged**